bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1519,12 +1529,16 @@ uint32_t PreProcessor::launchReqPreProcessing(const PreProcessRequestMsgSharedPt
       preProcessReqMsg->requestBuf(),
       std::string(preProcessReqMsg->requestSignature(), preProcessReqMsg->requestSignatureLength()),
       maxPreExecResultSize_,
-      (char *)getPreProcessResultBuffer(clientId, reqSeqNum, reqOffsetInBatch)});
-  accumulatedRequests.back().blockId = preProcessReqMsg->primaryBlockId();
+      preProcessResultBuffer});
   requestsHandler_.execute(accumulatedRequests, std::nullopt, cid, span);
   const IRequestsHandler::ExecutionRequest &request = accumulatedRequests.back();
   const auto status = request.outExecutionStatus;
-  const auto resultLen = request.outActualReplySize;
+  // Append the conflict detection block id and add sizeof(uint64_t) the resulting length.
+  memcpy(preProcessResultBuffer + request.outActualReplySize, reinterpret_cast<char *>(&blockId), sizeof(uint64_t));
+  const auto resultLen = request.outActualReplySize + sizeof(uint64_t);
+  LOG_INFO(logger(),
```

Comment:
Do we want to have this hash calculation happen always? We need it only for a log.

Reply:
it should be on debug level, thanks